Scripts/RemediationScripts/Remediate-StopExposingServiceFabricReverseProxyPort.ps1

```diff
@@ -0,0 +1,956 @@
+        3. Execute the script to stop the Reverse porxy port to be exposed on Service Fabric(s) in the Subscription. Refer `Examples`, below.
+
+    To roll back:
+        1. Download the script.
+        2. Load the script in a PowerShell session. Refer https://aka.ms/AzTS-docs/RemediationscriptExcSteps to know more about loading the script.
+        3. Execute the script to expose the Reverse proxy port publically on Service Fabric(s) in the Subscription. Refer `Examples`, below.
+
+# Examples:
+    To remediate:
+        1. To review the Service Fabric(s) in a Subscription that will be remediated:
+    
+           Disable-ReverseProxyPortForServiceFabric -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -DryRun
+
+        2. The Reverse proxy must not b exposed publically for Service Fabric(s) in the Subscription:
+       
+           Disable-ReverseProxyPortForServiceFabric -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck
+
+        3. The Reverse proxy must not b exposed publically for Service Fabric(s) in the Subscription, from a previously taken snapshot:
+       
+           Disable-ReverseProxyPortForServiceFabric -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\DoNotExposeReverseProxyforServiceFabric\ServiceFabricWithReverseProxyExposed.csv
+
+        To know more about the options supported by the remediation command, execute:
+        
+        Get-Help Disable-ReverseProxyPortForServiceFabric -Detailed
+
+    To roll back:
+        1. Expose the Reverse proxy port publically for Service Fabric(s) in the Subscription, from a previously taken snapshot:
+           Enable-ReverseProxyPortForServiceFabric -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\DoNotExposeReverseProxyPortforServiceFabric\RemediatedServiceFabric.csv
+       
+        To know more about the options supported by the roll back command, execute:
+        
+        Get-Help Enable-ReverseProxyPortForServiceFabric -Detailed        
+###>
+
+
+function Setup-Prerequisites
+{
+    <#
+        .SYNOPSIS
+        Checks if the prerequisites are met, else, sets them up.
+
+        .DESCRIPTION
+        Checks if the prerequisites are met, else, sets them up.
+        Includes installing any required Azure modules.
+
+        .INPUTS
+        None. You cannot pipe objects to Setup-Prerequisites.
+
+        .OUTPUTS
+        None. Setup-Prerequisites does not return anything that can be piped and used as an input to another command.
+
+        .EXAMPLE
+        PS> Setup-Prerequisites
+
+        .LINK
+        None
+    #>
+
+    # List of required modules
+    $requiredModules = @("Az.Accounts", "Az.ServiceFabric")
```

Comment:
Get-AzVmss command is in Az.Compute module add that as well here.

Reply:
Get-AzVirtualNetwork & Get-AzVirtualNetworkGateway are in Az.Netwrok module.

Reply:
Hi Tarun, this has been resolved.